native/Avalonia.Native/src/OSX/platformthreading.mm

```diff
@@ -138,6 +138,12 @@ virtual void Signal(int priority) override
             return new TimerWrapper(callback, ms);
         }
     }
+    
+    virtual HRESULT TerminateApp () override
+    {
+        [NSApp stop:0];
```

Comment:
This line indicates everything went well but in fact you did an emergency shutdown of the app because of an unrecoverable state.